sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/EventProcessor.cs

```diff
@@ -0,0 +1,230 @@
+        public string Name { get; }
+
+        /// <summary>
+        ///   The client used to interact with the Azure Event Hubs service.
+        /// </summary>
+        ///
+        private EventHubClient InnerClient { get; }
+
+        /// <summary>
+        ///   The name of the consumer group this event processor is associated with.  Events will be
+        ///   read only in the context of this group.
+        /// </summary>
+        ///
+        private string ConsumerGroup { get; }
+
+        /// <summary>
+        ///   An instance of a class that implements the <see cref="IPartitionProcessorFactory" /> interface.
+        ///   It's provided by the user and it's used to create partition processors.
+        /// </summary>
+        ///
+        private IPartitionProcessorFactory PartitionProcessorFactory { get; }
+
+        /// <summary>
+        ///   An instance of a class that implements the <see cref="IPartitionManager" /> interface.
+        ///   It's provided by the user and it's used to interact with the storage system, dealing with
+        ///   leases and checkpoints.
+        /// </summary>
+        ///
+        private IPartitionManager PartitionManager { get; }
+
+        /// <summary>
+        ///   The set of options to use for this event processor.
+        /// </summary>
+        ///
+        private EventProcessorOptions Options { get; }
+
+        /// <summary>
+        ///   A <see cref="CancellationTokenSource"/> instance to signal the request to cancel the current running task.
+        /// </summary>
+        ///
+        private CancellationTokenSource RunningTaskTokenSource { get; set; }
+
+        /// <summary>
+        ///   The set of partition pumps used by this event processor.  Partition ids are used as keys.
+        /// </summary>
+        ///
+        private ConcurrentDictionary<string, PartitionPump> PartitionPumps { get; set; }
+
+        /// <summary>
+        ///   The running task responsible for checking the status of the owned partition pumps.
+        /// </summary>
+        ///
+        private Task RunningTask { get; set; }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="EventProcessor"/> class.
+        /// </summary>
+        ///
+        /// <param name="eventHubClient">The client used to interact with the Azure Event Hubs service.</param>
+        /// <param name="consumerGroup">The name of the consumer group this event processor is associated with.  Events are read in the context of this group.</param>
+        /// <param name="partitionProcessorFactory">A factory used to create partition processors.  Its implementation must be provided by the user.</param>
+        /// <param name="partitionManager">A partition manager used to interact with the storage system.  Its implementation must be provided by the user.</param>
+        ///
+        public EventProcessor(EventHubClient eventHubClient,
+                              string consumerGroup,
```

Comment:
I believe we decided to flip the order here so that `consumerGroup` is first. Then from a user perspective it would be easier to understand the intent

> I want to receive events from this consumer group using this event hub client and processors from this factory with checkpointing done using this partition manager

Reply:
Omg, I didn't know the order was a thing. Just updated it. Are there any other cases like this I should be aware of?